### ISSUE TITLE
pView loses the ability to bring bView to front after a while

### ISSUE THREAD

**Issue by damithc** [HubTurbo/HubTurbo#862]
Noticed in both Win8 and Win7.
It works fine for a while, and then it loses the ability to bring bView to front when pView gets focus.

**Comment by damithc** [comment]
May be the fix for #810 is the culprit as I did not have this issue before that.

**Comment by acruis** [comment]
Yeah, that's most likely the reason why. I'm not that familiar with the Windows API, or the JNA implementation of it. I can continue to take some time and investigate this issue, or revert the fix for now. Would you like that?

**Comment by acruis** [comment]
Also, if you encounter this issue again, it would be great if you could tell me the conditions of the two windows when it happened. (such as whether bView was minimized beforehand, etc)

**Comment by damithc** [comment]
I don't think reverting is an option. Remember, prior to that fix HT was completely unusable for some users. The current problem is annoying but can be overcome by restarting.

**Comment by damithc** [comment]
Doesn't seem to be related to minimizing bView. I tried minimizing a few times but it did not break anything. It just seem to happen after a while. Yet to observe anything that may be causing it.

**Comment by acruis** [comment]
Noted. Do keep an eye out still for things that might cause the issue :)

**Comment by damithc** [comment]
Perhaps we can revert to the previous way of doing things but prevent chrome window from blocking the pview entirely. The problem Hong Jin previously had can be avoided if the chrome window is not maximized right?

**Comment by acruis** [comment]
Yes. The problem is that different people have different workflows. People will tend to maximize the Chrome window if they don't have the Chrome window on a separate monitor from the pView (it prevented Hong Jin from viewing PR diffs fully and he had to scroll horizontally to see the diffs, it was very annoying).

We can prevent the Chrome window from blocking the pView by resizing it when pView gains focus, and then trigger the infinite loop like what we did before. Do you want to implement that?

**Comment by damithc** [comment]
What do we do if the pView is maximized and it gains focus?
Let's discuss when we meet tomorrow.

**Comment by damithc** [comment]
Even when the pView cannot bring bView to front, bView still gets focus if I do an action like press 'M'. Not sure if that's useful for figuring out a solution.

**Comment by acruis** [comment]
What do you mean by bView getting focus?

**Comment by damithc** [comment]
It comes to the front. But I guess that's because it has focus which is
normal Windows behavior.

**Comment by acruis** [comment]
After it gets focus, is pView able to bring bView to front?

**Comment by damithc** [comment]
No (as far as I can remember). I'll double-check next time it happens.

**Comment by damithc** [comment]
Another interesting observation. pView seem to be able to regain the ability to bring back the bView to front. Today morning it was broken, but now it seems to be working. I did not restart HT during that interval. :-p

**Comment by acruis** [comment]
Noted. Do tell me if you have more observations you can make :-)

**Comment by damithc** [comment]
@rd1992 you working on this?

**Comment by rd1992** [comment]
This will be fixed with #1005. Working on that now
